src/scripts/components/notification.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Single Notification UI component
+ * https://github.com/WordPress/wp-feature-notifications/issues/16#issuecomment-896031592
+ * https://github.com/WordPress/wp-feature-notifications/issues/37#issuecomment-896080025
+ */
+import { __ } from '@wordpress/i18n';
+
+// Register components
+import { Component } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import NoticeImage from './noticeImage';
+
+import moment from 'moment';
+import { delay } from '../utils';
+
+// Notice class method.
+export default class Notice extends Component {
+	/**
+	 * Setting the default props for the component.
+	 *
+	 * @typedef Notice
+	 * @property {string}   title         - The Notice title
+	 * @property {string}   message       - The notice message
+	 * @property {string}   acceptLink    - The notice Href
+	 * @property {string}   acceptMessage - The notice "Accept" message
+	 * @property {string}   dismissLabel  - The notice "Dismiss" message
+	 * @property {string}   source        - The notice source
+	 * @property {string}   severity      - The notice severity
+	 * @property {boolean}  unread        - the notice is unread
+	 * @property {boolean}  dismissible   - if this notice is dismissible
+	 * @property {Function} onDismiss     - Unmount this notice
+	 */
+	static defaultProps = {
+		title: undefined,
+		message: undefined,
+		acceptMessage: __( 'Accept' ),
+		acceptLink: '#',
+		dismissLabel: __( 'Dismiss' ),
+		source: 'WordPress',
+		dismissible: false,
+		onDismiss: () => async () => {
+			delay( 100 ).then( () => this.unmount );
+		},
+	};
+
+	render() {
+		const classes = [
+			'wp-notification',
+			'wp-notice-' + this.props.id,
+			this.props.dismissible ? 'is-dismissible' : null,
+			this.props.severity || null,
+			this.props.unread ? 'unread' : null,
+			this.props.additionalClassName,
+		].filter( Boolean );
+
+		return (
```

Comment:
This is good, but it is a bit of a massive thing 😅 

Maybe not now, but it would be good to split it up slightly, so something like this for each of the individual conditional components.

```js

const dashboardActions = () => <something>Some buttons</something>

return (
  <div>
    { (location === 'dashboard') && dashboardActions }
  </div>
)

Reply:
Following this idea I also created a component for the actions (ok+dismiss button), but I was wondering if it would be a good idea to do the same for the title/text. What do you think?